app/src/containers/SearchPage/actions.js

```diff
@@ -1,30 +1,65 @@
-import {
-  FETCH_MOVIES_REQUEST,
-  FETCH_MOVIES_FAILURE,
-  FETCH_MOVIES_SUCCESS,
-  SET_MOVIES_QUERY,
-  SET_MOVIES_SORT_BY,
+// @flow
+
+import moviesService from '../../services/MoviesService';
+import type { Movie } from './reducer';
+import { 
+  FETCH_MOVIES_FAILURE, 
+  FETCH_MOVIES_REQUEST, 
+  FETCH_MOVIES_SUCCESS, 
+  SET_MOVIES_QUERY, 
+  SET_MOVIES_SORT_BY, 
   SET_MOVIES_SEARCH_BY,
 } from './action-types';
-import moviesService from '../../services/MoviesService';
 
-function getMoviesRequest() {
+type FetchMoviesRequestAction = {| 
+  +type: typeof FETCH_MOVIES_REQUEST 
+|};
+type FetchMoviesFailureAction = {| 
+  +type: typeof FETCH_MOVIES_FAILURE, 
+  +payload: string 
+|};
+type FetchMoviesSuccessAction = {| 
+  +type: typeof FETCH_MOVIES_SUCCESS, 
+  +payload: { 
+    +movies: Array<Movie>,
+    +moviesCount: number
+  } 
+|};
+export type SetMoviesQueryAction = {|
```

Comment:
You will not need to export these types:
```javascript
type SetMoviesQueryAction = {|
  +type: typeof SET_MOVIES_QUERY, 
  +payload: string 
|};
type SetMoviesSortByAction = {|
  +type: typeof SET_MOVIES_SORT_BY, 
  +payload: string 
|};
type SetMoviesSearchByAction = {|
  +type: typeof SET_MOVIES_SEARCH_BY, 
  +payload: string 
|};
```

Reply:
Because you will use "typeof" instead.

Reply:
Neat! Thanks.